### PR TITLE
fix: Ensure patched merge-source-map is included in package

### PR DESCRIPTION
**Pull request by chrisbreiding**
### User facing changelog

N/A - only affects `@cypress/webpack-preprocessor`

### Additional details

In https://github.com/cypress-io/cypress/pull/24367, I used patch-package to patch an issue in `merge-source-map`. Turns out patches don't get applied when a user installs the package. I should have heeded [Emily's concerns about this possibility](https://github.com/cypress-io/cypress/pull/24367#issuecomment-1292053610).

I'm going with her suggestion to just own the utility ourselves by copying and attributing the code.

### PR Tasks

- [N/A] Have tests been added/updated?
- [N/A] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [N/A] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [N/A] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: npm/webpack-preprocessor/package.json [hunk 1]
```diff
@@ -27,8 +26,8 @@
     "loader-utils": "^2.0.0",
     "lodash": "^4.17.20",
     "md5": "2.3.0",
-    "merge-source-map": "1.1.0",
-    "patch-package": "6.4.7",
+    "source-map": "^0.6.1",
+    "webpack": "^5.39.0",
```
Comment (emilyrohrbough) [inline npm/webpack-preprocessor/package.json]:
```suggestion
```

Comment (chrisbreiding) [reply npm/webpack-preprocessor/package.json]: Good catch. It's pretty annoying that running the tests changes the dependencies. I'll have to look into fixing that.

Addressed in https://github.com/cypress-io/cypress/pull/24490/commits/a8d84076159a2966eb7ec9ab01396a09aca8002e.

Comment (emilyrohrbough) [reply npm/webpack-preprocessor/package.json]: ohh I didn't even realize thats where it was coming from! that's weird.

**Review by emilyrohrbough — APPROVED** [review]

File: npm/webpack-preprocessor/package.json (hunk 1 above)
Comment (chrisbreiding) [reply npm/webpack-preprocessor/package.json]: Yeah, for some of the tests, it installs webpack 5 and runs against it. There must be a way to make that ephemeral and not change the source package.json though.

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**